### ISSUE TITLE
Not getting any predictions on custom yolo v8 model

### ISSUE THREAD

**Issue by cheenuz** [ultralytics/ultralytics#1812]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

Regards!

I am using yolo v8 with my rtx 3050 on python.
I had no problems with the coco128 dataset, and I am getting no issue in training it and getting acuratte prediction.
But when I try to load my custom dataset, I can do the training normally but It just gives me the same input image when I try to make a prediction. I am trying to make a model to classify masks on people, with only 2 classes : no mask and mask.

I have over 800 images to train, so lack of data isnt the issue.
I am using transfer learning to retrain the medium model.

There isn't a problem with my data, because the label image which is stored contains accurate data on the masks and no masks.
whereas the prediction image fails to do anything.

### Additional

![val_batch0_labels](https://user-images.githubusercontent.com/129872751/229831687-149cfab9-6ae8-4c0e-8589-7ae47ab457f0.jpg)
![val_batch0_pred](https://user-images.githubusercontent.com/129872751/229831708-0da71bf0-7721-4b2f-b664-8368fa3c85b7.jpg)
![confusion_matrix](https://user-images.githubusercontent.com/129872751/229831902-69c104f0-ff89-4224-ae1a-6bf366ce13bf.png)

Here are my label files and predicted files. You can see the predicted files are blank.

**Comment by glenn-jocher** [comment]
Hi @jc-wan, sorry to hear you are experiencing issues with custom training. From your description it sounds like your model is not learning to predict your custom classes. A couple of things I would suggest checking: 
1. Ensure that your annotations are in the correct format, for example in YOLO format should have one txt file for each image with each row containing: [class index, x_center, y_center, width, height]. If you can see the annotations on the images themselves, perhaps check that object locations are correct and their classes are properly labeled. 
2. Consider increasing the learning rate and try again.  
3. You may try a simpler model architecture with fewer layers and a lower number of filters to see if your custom training data can be learned more easily. Lastly, make sure you have enough training data and that your validation dataset is a representative sample of your full dataset. 

Hope this helps, let me know how it goes!

**Comment by utkuyazici1** [comment]
Hi, I have faced with the same issue, could you find any solution?

**Comment by pderrenger** [comment]
Hello @utkuyazici1! A couple of general troubleshooting steps you may consider:
1. Ensure that your classes are properly labeled in the custom dataset, and the annotations are in the correct format.
2. Try increasing the learning rate and repeat the training process to see if it helps the model better learn your custom classes.
3. Consider using a simpler model architecture with fewer layers, or try adjusting the model's configuration to better fit your custom dataset.

These steps might help in resolving the issue. Let me know if you have any further questions or if there's anything else I can assist you with!

**Comment by dpant2426** [comment]
Hi, I hope you have found a solution by now. Can you share what the issue was? I'm facing a similar problem.
Thanks

**Comment by glenn-jocher** [comment]
Hi! If you're still facing issues, ensure your dataset is correctly formatted and annotations are accurate. Adjusting the learning rate or model architecture might also help. Let me know if you need further assistance!
